src/plugins/presentation_util/public/components/saved_object_save_modal_dashboard.tsx

```diff
@@ -9,145 +9,64 @@
-              <EuiFlexItem grow={false}>
-                <EuiIconTip
-                  type="iInCircle"
-                  content={
-                    <FormattedMessage
-                      id="presentationUtil.saveModalDashboard.dashboardInfoTooltip"
-                      defaultMessage="Items added to a dashboard will not appear in the library and must be edited from the dashboard."
-                    />
-                  }
-                />
-              </EuiFlexItem>
-            </EuiFlexGroup>
-          }
-          hasChildLabel={false}
-        >
-          <EuiPanel color="subdued" hasShadow={false} data-test-subj="add-to-dashboard-options">
-            <div>
-              <EuiRadio
-                checked={dashboardOption === 'existing'}
-                id="existing-dashboard-option"
-                name="dashboard-option"
-                label={i18n.translate(
-                  'presentationUtil.saveModalDashboard.existingDashboardOptionLabel',
-                  {
-                    defaultMessage: 'Existing',
-                  }
-                )}
-                onChange={() => setDashboardOption('existing')}
-                disabled={isDisabled}
-              />
-
-              <div className="savAddDashboard__searchDashboards">
-                <DashboardPicker
-                  savedObjectsClient={savedObjectsClient}
-                  isDisabled={dashboardOption !== 'existing'}
-                  onChange={(dash) => {
-                    setSelectedDashboard(dash);
-                  }}
-                />
-              </div>
-
-              <EuiSpacer size="s" />
-
-              <EuiRadio
-                checked={dashboardOption === 'new'}
-                id="new-dashboard-option"
-                name="dashboard-option"
-                label={i18n.translate(
-                  'presentationUtil.saveModalDashboard.newDashboardOptionLabel',
-                  {
-                    defaultMessage: 'New',
-                  }
-                )}
-                onChange={() => setDashboardOption('new')}
-                disabled={isDisabled}
-              />
-
-              <EuiSpacer size="s" />
-
-              <EuiRadio
-                checked={dashboardOption === null}
-                id="add-to-library-option"
-                name="dashboard-option"
-                label={i18n.translate('presentationUtil.saveModalDashboard.libraryOptionLabel', {
-                  defaultMessage: 'No dashboard, but add to library',
-                })}
-                onChange={() => setDashboardOption(null)}
-                disabled={isDisabled}
-              />
-            </div>
-          </EuiPanel>
-        </EuiFormRow>
-      </>
-    );
-  };
+  const rightOptions = !disableDashboardOptions
+    ? () => (
+        <SaveModalDashboardSelector
+          onSelect={(dash) => {
+            setSelectedDashboard(dash);
+          }}
+          {...{ copyOnSave, documentId }}
+        />
```

Comment:
Some callback needs to be passed it for selecting the dashboard option. At the moment, the state is being modified in the `SaveModalDashboardSelector` but it doesn't propagate up to this component

Reply:
I'll take a look at this.

Reply:
Fixed!